lib/Ceilometer/Types.hs

```diff
@@ -1 +1,111 @@
+{-# LANGUAGE OverloadedStrings #-}
+{-# LANGUAGE RankNTypes        #-}
+
 module Ceilometer.Types where
+
+--import           Control.Isomorphism.Partial.Iso
+import           Data.Bimap                      (Bimap)
+import qualified Data.Bimap                      as B
+import           Data.HashMap.Strict             (HashMap)
+import qualified Data.HashMap.Strict             as H
+import           Data.Text                       (Text)
+import qualified Data.Text                       as T
+import           Data.Word
+
+import           Marquise.Client
+import           Vaultaire.Types
+
+import           Ceilometer.Types.Base
+import           Ceilometer.Types.IP
+import           Ceilometer.Types.Volume
+
+import Control.Lens.Prism
+
+type Env = (FlavorMap, SourceDict)
+
+lookupMetricName = lookupSource "metric_name"
+lookupEvent      = lookupSource "_event"
+
+type CSimple = SimplePoint
+
+-- | Views the raw @SimplePoint@ as-is.
+cSimple :: Env -> Prism' SimplePoint CSimple
```

Comment:
prism' id f seems more like a Fold to me unless you're actually planning to do something in the backwards direction.

What's going on here is a little misleading. I don't like the idea of a type synonym here, perhaps a newtype? If all you want to do is have a partial composable reviewable thing, you may be looking for a Fold.

```haskell
cSimple :: Env -> Fold SimplePoint CSimple
csSimple (_, sd) = folding parse
  where ...
```

Then you can review it, because that works on folds.